PocketKit/Sources/Analytics/AppEvents/Collection.swift

```diff
@@ -0,0 +1,240 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+
+public extension Events {
+    struct Collection {}
+}
+
+public extension Events.Collection {
+    /**
+     * Fired when a user views the collection screen
+     */
+    static func screenView() -> Impression {
+        return Impression(
+            component: .screen,
+            requirement: .instant,
+            uiEntity: UiEntity(
+                .screen,
+                identifier: "collection.screen"
+            )
+        )
+    }
+
+    // MARK: Tracking Navbar Actions
+    /**
+     * Fired when a user Saves a card on a collection
+     */
+    static func saveClicked(url: String) -> Engagement {
+        return Engagement(
+            .save(contentEntity: ContentEntity(url: url)),
+            uiEntity: UiEntity(
+                .button,
+                identifier: "collection.save"
+            )
+        )
+    }
+
+    /**
+     * Fired when a user archives / unsaves a card on a collection
+     */
+    static func unsaveClicked(url: String) -> Engagement {
+        return Engagement(
+            .save(contentEntity: ContentEntity(url: url)),
+            uiEntity: UiEntity(
+                .button,
+                identifier: "collection.unsave"
+            )
+        )
+    }
+
+    /**
+     Fired when a user taps on the save icon in the collection nav bar and moves item from archive to save
+     */
+    static func moveFromArchiveToSavesClicked(url: String) -> Engagement {
+        return Engagement(
+            .save(contentEntity: ContentEntity(url: url)),
+            uiEntity: UiEntity(
+                .button,
+                identifier: "collection.un-archive"
```

Comment:
Just leaving a comment for anyone else reading - this is correct, and should be hyphenated according to our living analytics document.

Reply:
thank you!